src/main/java/com/epam/spring/library/dto/mapper/impl/LanguageMapperImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.epam.spring.library.dto.mapper.impl;
+
+import com.epam.spring.library.dto.mapper.LanguageMapper;
+import com.epam.spring.library.model.Language;
+import com.epam.spring.library.repository.LanguageRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class LanguageMapperImpl implements LanguageMapper {
+    private final LanguageRepository repository;
+
+    @Override
+    public Language toLanguage(String code) {
+        if (code == null) {
+            return null;
+        }
+
+        return repository.getLanguage(code);
+    }
+
+    @Override
+    public String toDTO(Language language) {
+        if (Objects.isNull(language)) {
+            return null;
+        }
+
+        return language.getCode();
+    }
+
+    @Override
+    public List<String> toDTO(List<Language> languages) {
+        if (Objects.isNull(languages)) {
+            return new ArrayList<>();
```

Comment:
This will create a new instance of `ArrayList` every time, which is not good.
Use `Collections.emptyList()` or `List.of()`